addons/hr/models/hr_plan.py

```diff
@@ -59,6 +82,142 @@ class HrPlan(models.Model):
     _name = 'hr.plan'
     _description = 'plan'
 
+    def _default_trigger_onboarding(self):
+        if self.plan_type == 'onboarding' and self.trigger:
+            return self.trigger
+        return 'manual'
+
+    def _default_trigger_offboarding(self):
+        if self.plan_type == 'offboarding' and self.trigger:
+            return self.trigger
+        return 'manual'
+
     name = fields.Char('Name', required=True)
     plan_activity_type_ids = fields.Many2many('hr.plan.activity.type', string='Activities')
     active = fields.Boolean(default=True)
+    plan_type = fields.Selection(
+        [
+            ('onboarding', 'Onboarding'),
+            ('offboarding', 'Offboarding'),
+            ('other', 'Other'),
+        ],
+        string='Type',
+        default='onboarding',
+        required=True,
+    )
```

Comment:
This way to define the doc is more readable:
```suggestion
    plan_type = fields.Selection([
            ('onboarding', 'Onboarding'),
            ('offboarding', 'Offboarding'),
            ('other', 'Other'),
        ], string='Type', default='onboarding', required=True)
```